freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/verticalscale/handler/FreeIpaVerticalScaleHandler.java

```diff
@@ -0,0 +1,78 @@
+
+import com.sequenceiq.cloudbreak.cloud.CloudConnector;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.handler.CloudPlatformEventHandler;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResourceStatus;
+import com.sequenceiq.cloudbreak.cloud.model.ResourceStatus;
+import com.sequenceiq.freeipa.flow.freeipa.verticalscale.FreeIpaVerticalScaleService;
+import com.sequenceiq.freeipa.flow.freeipa.verticalscale.event.FreeIpaVerticalScaleRequest;
+import com.sequenceiq.freeipa.flow.freeipa.verticalscale.event.FreeIpaVerticalScaleResult;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class FreeIpaVerticalScaleHandler implements CloudPlatformEventHandler<FreeIpaVerticalScaleRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FreeIpaVerticalScaleHandler.class);
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Inject
+    private FreeIpaVerticalScaleService verticalScale;
+
+    @Override
+    public Class<FreeIpaVerticalScaleRequest> type() {
+        return FreeIpaVerticalScaleRequest.class;
+    }
+
+    @Override
+    public void accept(Event<FreeIpaVerticalScaleRequest> event) {
+        LOGGER.debug("Received event: {}", event);
+        FreeIpaVerticalScaleRequest<FreeIpaVerticalScaleResult> request = event.getData();
+        CloudContext cloudContext = request.getCloudContext();
+        com.sequenceiq.freeipa.api.v1.freeipa.stack.model.scale.FreeIpaVerticalScaleRequest freeIPAVerticalScaleV1Request
+                = request.getFreeIPAVerticalScaleRequest();
+        try {
+            CloudConnector connector = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = connector.authentication().authenticate(cloudContext, request.getCloudCredential());
+            List<CloudResourceStatus> resourceStatus = verticalScale.verticalScale(ac, request, connector);
+            LOGGER.info("Vertical scaling resource statuses: {}", resourceStatus);
+            FreeIpaVerticalScaleResult result = new FreeIpaVerticalScaleResult(
+                    request.getResourceId(),
+                    ResourceStatus.UPDATED,
+                    resourceStatus,
+                    freeIPAVerticalScaleV1Request);
+            request.getResult().onNext(result);
+            eventBus.notify(result.selector(), new Event<>(event.getHeaders(), result));
+            LOGGER.debug("Vertical scaling successfully finished for {}, and the result is: {}", cloudContext, result);
+        } catch (Exception e) {
+            LOGGER.error("Vertical scaling stack failed", e);
+            FreeIpaVerticalScaleResult result = new FreeIpaVerticalScaleResult(
```

Comment:
this should be `FreeIpaVerticalScaleFailureEvent` or you have to add a converter to `verticalScaleFailedAction`